content/docs/user-guide/setup-google-drive-remote.md

```diff
@@ -1,9 +1,120 @@
+[connect using a custom Google Cloud project](#connect-using-a-custom-google-cloud-project),
+which puts you in control of these limits.
+
+Having your own GC project, it's also possible to
+[use a service account](#using-service-accounts) for automating tasks that need
+to establish GDrive remote connections (e.g. CI/CD).
+
+## URL formats to specify folder location
+
+There's a few alternatives to construct a GDrive remote URL for different uses,
+such as a folder or subfolder in root, shared folders not owned by your account,
+etc. The URL is formed with a _base_, and an optional _path_ to an **existing**
+folder i.e. `gdrive://<base>/path/to/folder`. The base can be one of:
+
+1. _Folder ID_ - unique identifier for each Google Drive folder
+
+   To obtain it, navigate to that folder in your web browser, and find it in the
+   address bar. For example, for
+   `https://drive.google.com/drive/folders/0AIac4JZqHhKmUk9PDA`:
+
+   ```dvc
+   $ dvc remote add gdfolder gdrive://0AIac4JZqHhKmUk9PDA
+   $ dvc remote add gdsubfolder \
+                  gdrive://0AIac4JZqHhKmUk9PDA/Data/text
+   ```
+
+   Note that [shared folders](https://support.google.com/drive/answer/7166529)
+   and [shared drives](https://support.google.com/a/users/answer/9310351) can
+   only be referenced this way.
+
+2. `root` - indicates your topmost Google Drive directory.
+
+   ⚠️ Only suitable for personal use, as sharing a remote configured this way
+   would cause DVC to try synchronizing data to/from different Google Drives for
+   every user.
+
+   ```dvc
+   $ dvc remote add mygdfolder gdrive://root/dvcstore
+   ```
+
+   > Although valid, we don't recommend using just `gdrive://root`, as it's not
+   > possible for more than one Google account to access a GDrive root ("My
+   > Drive").
+
+3. `appDataFolder` -
+   [special hidden folder](https://developers.google.com/drive/api/v2/appdata)
+   (unique per user) meant to store application-specific data. This is a good
+   choice to prevent accidentally deleting remote storage data from the Google
+   Drive web UI.
+
+   ```dvc
+   $ dvc remote add gdappata gdrive://appDataFolder
+   ```
+
+## Authorization
+
+On the first usage of a GDrive [remote](/doc/command-reference/remote), for
+example when trying to `dvc push` for the first time after adding the remote
+with a [valid URL](#url-format-to-specify-folder-location), DVC will prompt you
+to visit a special Google authorization web page. There you'll need to sign into
+your Google account. The
+[auth process](https://developers.google.com/drive/api/v2/about-auth) will ask
+you to grant DVC the necessary permissions, and produce a verification code
+needed for DVC to complete the connection. On success, this code will be cached
+in a Git-ignored directory located in `.dvc/tmp/gdrive-user-credentials.json`.
+
+⚠️ In order to prevent unauthorized access to your Google Drive, **do not share
+your verification code with others**. Each team member should go through this
```

Comment:
... don't share produced credentials file (not sure sharing verification code is a big deal)

Reply:
Good catch. Fixed!